plugins/inputs/github/github.go

```diff
@@ -191,6 +205,33 @@ func getFields(repositoryInfo *github.Repository) map[string]interface{} {
 	}
 }
 
+func (g *GitHub) getPullRequestFields(ctx context.Context, owner, repo string) (map[string]interface{}, error) {
+	options := github.SearchOptions{
+		TextMatch: false,
+		ListOptions: github.ListOptions{
+			PerPage: 100,
+			Page:    1,
+		},
+	}
+
+	classes := []string{"open", "closed"}
+	fields := make(map[string]interface{})
+	for _, class := range classes {
+		q := fmt.Sprintf("repo:%s/%s is:pr is:%s", owner, repo, class)
+		searchResult, response, err := g.githubClient.Search.Issues(ctx, q, &options)
+		if err != nil {
+			return fields, err
```

Comment:
It may happen that `fields` is not empty when err != nil. Do you still want to return it in case of error?